From 15a3380dba38351d1c3d480ae2321134481bb5fd Mon Sep 17 00:00:00 2001
From: Sangkyu Lee <sk82.lee@lge.com>
Date: Mon, 7 Jan 2013 10:31:53 +0900
Subject: [PATCH 253/283] Fix unexpected rotation change when re-enabling
 auto-rotate

getProposedRotation() method returns old value when re-enabling
auto-rotate option. So you can see unexpected rotation change
with the following steps.

settings -> display -> enable auto-rotate
-> rotate device to landscape -> disable auto-rotate
-> rotate device to portrait -> enable auto-rotate

This patch makes mSensorEventListener reset before it is registered again.

Bug: 7964531

Change-Id: I6a8d287bbd9809ef31de67c54efb885e9a4fd76f
Signed-off-by: Sangkyu Lee <sk82.lee@lge.com>
---
 core/java/android/view/WindowOrientationListener.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/java/android/view/WindowOrientationListener.java b/core/java/android/view/WindowOrientationListener.java
index 4c34dd4..bf77c67 100644
--- a/core/java/android/view/WindowOrientationListener.java
+++ b/core/java/android/view/WindowOrientationListener.java
@@ -98,6 +98,7 @@ public abstract class WindowOrientationListener {
             if (LOG) {
                 Log.d(TAG, "WindowOrientationListener enabled");
             }
+            mSensorEventListener.reset();
             mSensorManager.registerListener(mSensorEventListener, mSensor, mRate);
             mEnabled = true;
         }
-- 
1.8.1.1

